storage/innobase/log/log0files_finder.cc

```diff
@@ -298,11 +310,29 @@ Log_files_find_result log_files_find_and_analyze(
       log_file_header_reset_flag(file_header.m_log_flags,
                                  LOG_HEADER_FLAG_FILE_FULL);
     }
+    file_index++;
 #endif /* XTRABACKUP */
 
     file_headers.emplace_back(file.m_id, file_header);
   }
 
+#ifdef XTRABACKUP
+  if (missing_files.size() == file_sizes.size()) {
+    /* All files are missing. */
+    return Log_files_find_result::FOUND_NO_FILES;
+  }
+  /*
+   * Remove missing files from the list.
+   * Do it backwards to avoid reindexing. Eg.:
+   * files.erase(files.begin() + 0); - pos 1 will become pos 0
+   */
+  std::set<size_t>::iterator index = missing_files.end();
+  while (index != missing_files.begin()) {
+    --index;
+    file_sizes.erase(file_sizes.begin() + *index);
+  }
```

Comment:
then we can use std::remove_if() with matches to saved file.m_ids ?

Reply:
Done